src/main/scala/dpla/ingestion3/harvesters/QueryUrlBuilder.scala

```diff
@@ -60,12 +51,12 @@ class OaiQueryUrlBuilder extends QueryUrlBuilder with Serializable {
 
     /*
      * Set metadataPrefix param if:
-     *   - the verb allows a metadataPrefix argument AND
+     *   - metadataPrefix is defined
      *   - there is no resumption token (if an OAI call with a resumption token
      *     also contains a metadataPrefix, an OAI badArgument error is returned).
      */
-    if(verbsWithMetadataPrefix.contains(verb) && !resumptionToken.isDefined) {
-      urlParams.setParameter("metadataPrefix", metadataPrefix)
+    if(metadataPrefix.isDefined && !resumptionToken.isDefined) {
```

Comment:
A way to get Codacy to shut up about this:

`if (!resumptionToken.isDefined) 
    metadataPrefix.foreach(prefix => urlParams.setParameter("metadataPrefix", prefix))
`
Not sure it's worth it, though. In any case, this is an interesting blog: https://blog.protegra.com/2014/01/28/exploring-scala-options/